config/routes.rb

```diff
@@ -3,4 +3,12 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+
+  resources :dishes, only: [:show] do
+    resources :dish_ingredients, only: [:update]
```

Comment:
As mentioned above, you'd want to change that to :create instead of :update since you're creating the joins record. 

"/dishes/:id/dish_ingredients" is a fine path, but remember that you can use the controller parameter if you wanted to just nest `ingredients` under there so that it's `"/dishes/:id/ingredients" by doing:
```
resources :dishes only: :show do
     resources :ingredients, only: :create, controller: "dish_ingredients"
end
```